File: prepublish.js
#!/usr/bin/env node

const fs = require("fs")
const path = require("path")

try {
  const commonPath = path.join(
    __dirname,
    "..",
    "Cerys-Moon-of-Fulgora",
    "common.lua",
  )
  const content = fs.readFileSync(commonPath, "utf8")

  if (content.match(/DEBUG_\w+ = true/)) {
    console.error("Error: Debug flags must be disabled before publishing.")
    console.error("Please set all DEBUG_ variables to false in common.lua")
    process.exit(1)
  }
} catch (error) {
  console.error("Error checking debug flags:", error)
  process.exit(1)
}

File: prepublish.lua
local function ensure_debug_flags_disabled()
	local common_path = "Cerys-Moon-of-Fulgora/common.lua"
	local file = io.open(common_path, "r")
	if not file then
		error("Could not open common.lua")
	end

	local content = file:read("*all")
	file:close()

	-- Check if any debug flags are true
	local debug_enabled = content:match("DEBUG_%w+ = true")
	-- if not debug_enabled then
	if debug_enabled then
		error("Debug flags must be disabled before publishing.")
	end
end

ensure_debug_flags_disabled()
